pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/ProcessRuntime.java

```diff
@@ -131,6 +134,19 @@
     @Override
     public void start() {
         java.lang.Runtime.getRuntime().addShutdownHook(new Thread(() -> process.destroy()));
+
+        // Note: we create the expected log folder before the function process logger attempts to create it
+        // This is because if multiple instances are launched they can encounter a race condition creation of the dir.
+        log.info("Creating function log directory {}", funcLogDir);
+        boolean success = createFolder(funcLogDir);
+
+        if (!success) {
```

Comment:
So what if the directory is already there? Shouldn't throw exception because directory is already there

Reply:
I think this is fine since new File(path).mkdirs() is idempotent